x-pack/legacy/plugins/ml/public/jobs/new_job_new/pages/components/pick_fields_step/components/advanced_view/detector_list.tsx

```diff
@@ -108,9 +108,11 @@ export const DetectorList: FC<Props> = ({ isActive, onEditJob, onDeleteJob }) =>
                     detectorToString(d)
                   )}
                 </EuiFlexItem>
-                <EuiFlexItem grow={false} style={{ margin: '8px' }}>
-                  <Buttons index={i} />
-                </EuiFlexItem>
+                {isActive && (
+                  <EuiFlexItem grow={false} style={{ margin: '8px' }}>
```

Comment:
why not use a gutter size prop for EuiFlexGroup instead of inline styles?

Reply:
only this flex item in the list needs the margin adjusting.
I agree it would be preferable to not be using inline styles, however i could not find a better solution.
The contained `Button` component could have a negative margin but i thought that would be worse.